### PR TITLE
python3Packages.tsfresh: init at 0.20.1

### PR DESCRIPTION
**Pull request by mbalatsko**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mbalatsko** [comment]
Result of `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"` on x86_64-linux:
```
1 package built:
python310Packages.tsfresh
```

**Comment by mbalatsko** [comment]
@GrahamcOfBorg build python310Packages.tsfresh

**Comment by mbalatsko** [comment]
Dependency of https://github.com/NixOS/nixpkgs/issues/247089

File: pkgs/development/python-modules/tsfresh/default.nix [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  src = fetchFromGitHub {
+    owner = "blue-yonder";
+    repo = "tsfresh";
+    rev = "v${version}";
+    hash = "sha256-JmdP/6aTnuYsBRiRq9zZng3xNYhOdr9V8bp1trAv508=";
+  };
+
+  patches = [
+    # The pyscaffold is not a buil dependency but just a python project bootstrapping tool, so we do not need it
+    ./remove-pyscaffold.patch
```
Comment (natsukium) [inline pkgs/development/python-modules/tsfresh/default.nix:42]: Can we use `substituteInPlace` instead?
The patch file makes updating a bit messy.

Maybe we could just make this line a blank line.
https://github.com/blue-yonder/tsfresh/blob/v0.20.1/setup.cfg#L34

File: pkgs/development/python-modules/tsfresh/default.nix [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    scikit-learn
+    tqdm
+    dask
+    distributed
+    stumpy
+    cloudpickle
+  ] ++ dask.optional-dependencies.dataframe;
+
+  nativeCheckInputs = [
+    pytestCheckHook
```
Comment (natsukium) [inline pkgs/development/python-modules/tsfresh/default.nix:62]:
```suggestion
  nativeCheckInputs = [
    pytest-xdist
    pytestCheckHook
```
We can enable parallel testing.
https://github.com/blue-yonder/tsfresh/blob/v0.20.1/setup.cfg#L72

File: pkgs/development/python-modules/tsfresh/default.nix [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+, seaborn
+, ipython
+, notebook
+, pandas-datareader
+}:
+
+buildPythonPackage rec {
+  pname = "tsfresh";
+  version = "0.20.1";
+  format = "setuptools";
```
Comment (natsukium) [inline pkgs/development/python-modules/tsfresh/default.nix:29]:
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/tsfresh/default.nix [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+  src = fetchFromGitHub {
+    owner = "blue-yonder";
+    repo = "tsfresh";
+    rev = "v${version}";
+    hash = "sha256-JmdP/6aTnuYsBRiRq9zZng3xNYhOdr9V8bp1trAv508=";
+  };
+
+  patches = [
+    # The pyscaffold is not a buil dependency but just a python project bootstrapping tool, so we do not need it
```
Comment (natsukium) [inline pkgs/development/python-modules/tsfresh/default.nix:41]:
```suggestion
    # The pyscaffold is not a build dependency but just a python project bootstrapping tool, so we do not need it
```

File: pkgs/development/python-modules/tsfresh/default.nix (hunk 1 above)
Comment (mbalatsko) [reply pkgs/development/python-modules/tsfresh/default.nix:43]: Should be better now, just removing one lin ;)

**Comment by natsukium** [comment]
@ofborg eval

**Review by natsukium — APPROVED** [review]
I'll merge it after ofborg's evaluation.

**Pull request merged**